### PR TITLE
Add Task class

### PR DESCRIPTION
**Pull request by johnrhimawan**
Previously, the ```Internship``` class did not contain ```Task```. Let's add a new class ```Task``` to encapsulate the interviews and tests that the user might attend.

As a user applying for internships, I would like to add my upcoming tasks such as interview dates and online assessment dates to keep track of my deliverables for the application.

The ```Task``` class contains the fields ```taskName``` and ```taskTime```.

### CODE REVIEW

File: src/main/java/seedu/intrack/logic/parser/TaskCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import seedu.intrack.logic.commands.TaskCommand;
+import seedu.intrack.logic.parser.exceptions.ParseException;
+import seedu.intrack.model.internship.Task;
+
+/**
+ * Parses input arguments and creates a new {@code TaskCommand} object
+ */
+public class TaskCommandParser implements Parser<TaskCommand> {
+    private static final Pattern TASK_COMMAND_FORMAT =
+            Pattern.compile("(?<index>\\d+)\\s+(?<description>.*)\\s+/at(?<dateTime>.*)");
```
Comment (jetlfj) [inline src/main/java/seedu/intrack/logic/parser/TaskCommandParser.java:20]: Would be good to provide a comment explaining what this regex means so other collaborators can make use of it more easily.

File: src/main/java/seedu/intrack/logic/commands/TaskCommand.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import seedu.intrack.model.Model;
+import seedu.intrack.model.internship.Internship;
+import seedu.intrack.model.internship.Task;
+
+/**
+ * Updates the current task of an Internship.
+ */
+public class TaskCommand extends Command {
+
+    public static final String COMMAND_WORD = "task";
```
Comment (jetlfj) [inline src/main/java/seedu/intrack/logic/commands/TaskCommand.java:21]: Are we using `task` or `addtask`?

File: src/main/java/seedu/intrack/logic/commands/TaskCommand.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Updates the task of the internship identified by "
+            + "the index number used in the displayed list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + " TASK\n"
+            + "Example: " + COMMAND_WORD + " 1 Technical Interview /at 04-11-2022(17:00)";
+
+    public static final String TASK_COMMAND_CONSTRAINTS = "TASK must be in the format: \n"
+            + COMMAND_WORD + " INDEX DESCRIPTION /at TIME\n"
+            + "TIME must be in the format dd-MM-yyyy(HH:mm)";
```
Comment (jetlfj) [inline src/main/java/seedu/intrack/logic/commands/TaskCommand.java:31]: I think the brackets are unnecessary and could be a hassle for users to type out?
Perhaps a simpler format like `dd-MM-yyyy HH:mm` could be used?

**Review by jetlfj — COMMENTED** [review]
Well done adding the new class and its methods! Just some minor edits recommended. Otherwise, LGTM!

File: src/main/java/seedu/intrack/logic/commands/TaskCommand.java (hunk 2 above)
Comment (johnrhimawan) [reply src/main/java/seedu/intrack/logic/commands/TaskCommand.java:21]: Thank you for pointing that out, I have changed it to ```addtask``` to make it more clear that it is for adding a task.

File: src/main/java/seedu/intrack/logic/parser/TaskCommandParser.java (hunk 1 above)
Comment (johnrhimawan) [reply src/main/java/seedu/intrack/logic/parser/TaskCommandParser.java:20]: Excellent suggestion, I have already added it in now.

File: src/main/java/seedu/intrack/logic/commands/TaskCommand.java (hunk 3 above)
Comment (johnrhimawan) [reply src/main/java/seedu/intrack/logic/commands/TaskCommand.java:31]: Thank you for the advice, I have changed the format to be more convenient for users.

**Review by jetlfj — COMMENTED** [review]
LGTM!

**Pull request merged**